src/makerom/entry.s

```diff
@@ -4,21 +4,38 @@
 
 .text
 
-#if defined(__sgi) && !defined(AVOID_UB)
-/* IDO assembler workaround: The makerom tool in the N64 SDK was given the bss segment size as a const
+#if defined(__GNUC__) || defined(__sgi)
+/* Modern GCC or IDO */
+#define PAD_TO 0x60
+#define LA(reg, sym) la reg, sym
+#define BOOT_STACK_TOP sBootThreadStack + BOOT_STACK_SIZE
+#else
+/* EGCS */
```

Comment:
huh, it's kinda confusing that EGCS doesn't define `__GNUC__` for assembly files. What do you think of only `#if defined(__sgi)` or `#if defined(__sgi) || defined(NON_MATCHING)` for the first branch?

Reply:
Went with the latter